modules/sdk-core/src/bitgo/lightning/lightning.ts

```diff
@@ -114,4 +120,33 @@ export class Lightning implements ILightning {
       throw new Error(`error(s) parsing response body: ${errors}`);
     });
   }
+
+  public async decodeLnurlPay(lnurl: string): Promise<DecodedLnurlPayRequest> {
+    const url = decodeLnurl(lnurl);
+    const { body } = await request.get(url);
+    const decodedRes = decodeOrElse(LnurlPayResponse.name, LnurlPayResponse, body, (errors) => {
+      throw new Error(`error(s) parsing lnurl response: ${errors}`);
+    });
+    /**
+     * From https://github.com/fiatjaf/lnurl-rfc/blob/luds/06.md#pay-to-static-qrnfclink
+     * a payment dialog must include: Domain name extracted from LNURL query string.
+     */
+    const domain = new URL(url).hostname;
+
+    return {
+      ...decodedRes,
+      domain,
+    };
+  }
+
+  public async fetchLnurlPayInvoice(params: LnurlPayParams): Promise<string> {
+    const { callback, milliSatAmount, metadata } = params;
+    const { pr: invoice } = callback.includes('?')
+      ? await this.bitgo.get(callback + `&amount=${milliSatAmount}`).result()
+      : await this.bitgo.get(callback).query({ amount: milliSatAmount }).result();
```

Comment:
These need to use superagent get instead of bitgo get, currently giving me that `ApiResponseError: invalid response HMAC, possible man-in-the-middle-attack` error.